pilot/pkg/kube/inject/webhook_test.go

```diff
@@ -961,7 +961,7 @@ func mergeParamsIntoHelmValues(params *Params, vals string, t testing.TB) string
 	paramsVals := params.intoHelmValues()
 	for path, value := range paramsVals {
 		setStr := fmt.Sprintf("%s=%s", path, escapeHelmValue(value))
-		if err := strvals.ParseIntoString(setStr, valMap); err != nil {
+		if err := strvals.ParseInto(setStr, valMap); err != nil {
```

Comment:
k8s.io/helm changed the semantics of strvals.ParseIntoString() in recent versions. The previous version preserved the value's type. The newer version converts values into their string equivalent (e.g. true => "true") which breaks conditional template expansion in the unit tests.